src/metric_iterator.c

```diff
@@ -126,13 +123,14 @@ IndexIterator *NewMetricIterator(t_docId *ids_list, double *metric_list, Metric
   ri->ctx = mi;
   ri->type = METRIC_ITERATOR;
   ri->mode = MODE_SORTED;
-  ri->current = NewMetricResult();
 
   // If we interested in yielding score
   if (key_pp) {
-    mi->curValue = RS_NullVal(); // first dummy value
-    ri->current->additional = array_newlen(RSAdditionalValue, 1);
+    ri->current = NewMetricResult();
+    // ResultMetrics_Add(ri->current, NULL, RS_NewValue(RSValue_Undef));
     *key_pp = &ri->ownKey; // export own key address
+  } else {
+    ri->current = NewMetricResult();
```

Comment:
revert to `ri->current = NewMetricResult()` in line 129, and move `*key_pp = &ri->ownKey;` to the next `if` and delete this if -else